pokemongo_bot/cell_workers/sniper.py

```diff
@@ -232,6 +248,9 @@ def initialize(self):
         self.altitude = uniform(self.bot.config.alt_min, self.bot.config.alt_max)
         self.sources = [SniperSource(data) for data in self.config.get('sources', [])]
 
+        if not hasattr(self.bot,"sniper_cache"):
+            self.bot.sniper_cache = []
```

Comment:
You had a "self.cache" which is nice but not persistent. Every time the task runs, it reset it to []. This creates a bot-global cache which will persist through task re-inits.